src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -109,26 +113,15 @@ public bool SnippetShortcutExists_NonBlocking(string shortcut)
         public virtual bool ShouldFormatSnippet(SnippetInfo snippetInfo)
             => false;
 
-        private void PopulateSnippetCaches()
+        private async Task PopulateSnippetCacheAsync()
         {
-            Debug.Assert(_expansionManager != null);
-
-            var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+            using var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+            RoslynDebug.Assert(_expansionManager != null);
 
             // In Dev14 Update2+ the platform always provides an IExpansion Manager
-            var asyncExpansionManager = (IExpansionManager)_expansionManager;
+            var expansionManager = (IExpansionManager)_expansionManager;
```

Comment:
Should this assert we're on the UI thread prior to the cast?

Reply:
Nope, per DavidP this is machine translated and can be cast on any thread.